src/client.rs

```diff
@@ -28,8 +28,11 @@ const CLIENT_VERSION: &str = "0.0.1";
 struct ClientState {
     sequence:     u64,   // sequence number of requests
     response_ack: Option<u64>,  // last acknowledged response sequence number from server
+    last_req_action: Option<RequestAction>,   // last one we sent to server TODO: this is wrong;
+                                              // assumes network is well-behaved
     name:         Option<String>,
     in_game:      bool,
+    game_slot:    Option<String>,             // None iff. in_game is false
```

Comment:
if `in_game` is false when this is `None`, couldn't we simply have a function which checks if `game_slot.is_some()` to determine if we are in game?

Reply:
Yes, you're right. I'll fix that. Thanks!